internal/events/eventhandlers/send_receiver_wallets_sms_invitation_event_handler.go

```diff
@@ -0,0 +1,107 @@
+	service                services.SendReceiverWalletInviteServiceInterface
+}
+
+var _ events.EventHandler = new(SendReceiverWalletsSMSInvitationEventHandler)
+
+func NewSendReceiverWalletsSMSInvitationEventHandler(options SendReceiverWalletsSMSInvitationEventHandlerOptions) *SendReceiverWalletsSMSInvitationEventHandler {
+	s, err := services.NewSendReceiverWalletInviteService(
+		nil,
+		options.MessengerClient,
+		options.AnchorPlatformBaseSepURL,
+		options.Sep10SigningPrivateKey,
+		options.MaxInvitationSMSResendAttempts,
+		options.CrashTrackerClient,
+	)
+	if err != nil {
+		log.Fatalf("error instantiating service: %s", err.Error())
+	}
+
+	tm := tenant.NewManager(tenant.WithDatabase(options.DBConnectionPool))
+	tr := router.NewMultiTenantDataSourceRouter(tm)
+	tenantDBConnectionPool, err := db.NewConnectionPoolWithRouter(tr)
+	if err != nil {
+		log.Fatalf("error getting tenant DB Connection Pool: %s", err.Error())
+	}
+
+	return &SendReceiverWalletsSMSInvitationEventHandler{
+		tenantManager:          tm,
+		tenantDBConnectionPool: tenantDBConnectionPool,
+		service:                s,
+		crashTrackerClient:     options.CrashTrackerClient,
+	}
+}
+
+func (h *SendReceiverWalletsSMSInvitationEventHandler) Name() string {
+	return "SendReceiverWalletsSMSInvitationEventHandler"
+}
+
+func (h *SendReceiverWalletsSMSInvitationEventHandler) CanHandleMessage(ctx context.Context, message *events.Message) bool {
+	return message.Topic == events.ReceiverWalletNewInvitationTopic
+}
+
+func (h *SendReceiverWalletsSMSInvitationEventHandler) Handle(ctx context.Context, message *events.Message) {
+	dataJSON, err := json.Marshal(message.Data)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, fmt.Sprintf("[SendReceiverWalletsSMSInvitationEventHandler] could not marshal data: %v", message.Data))
+		return
+	}
+
+	var receiverWalletInvitationData []schemas.EventReceiverWalletSMSInvitationData
+	err = json.Unmarshal(dataJSON, &receiverWalletInvitationData)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, fmt.Sprintf("[SendReceiverWalletsSMSInvitationEventHandler] could not unmarshal data: %v", message.Data))
+		return
+	}
+
+	t, err := h.tenantManager.GetTenantByID(ctx, message.TenantID)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, "[SendReceiverWalletsSMSInvitationEventHandler] error getting tenant by id")
+		return
+	}
+
+	ctx = tenant.SaveTenantInContext(ctx, t)
+
+	models, err := data.NewModels(h.tenantDBConnectionPool)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, "[SendReceiverWalletsSMSInvitationEventHandler] error getting models")
+		return
+	}
+
+	h.service.SetModels(models)
+	if err := h.service.SendInvite(ctx, receiverWalletInvitationData...); err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, "[SendReceiverWalletsSMSInvitationEventHandler] sending receiver wallets invitation")
+		return
+	}
```

Comment:
🤔  Resurfacing @marwen-abid's [suggestion](https://github.com/stellar/stellar-disbursement-platform-backend/pull/133#discussion_r1446511958), the idea is to maintain the tenant logic outside the service, right?

Shouldn't you be passing the final data source here, rather than relying on the pair (context, mtnDataSource)?

Reply:
This is the event handler, not the service - this layer gets the message from the event broker and calls the service. Here's the place to get the tenant and pass the context through. WDYT?

Reply:
In the end, the business logic (the service) doesn't know about tenants.

Reply:
All good then! Can we create a reusable code that does a similar thing in the other handlers?